src/client/dfs/lookup.c

```diff
@@ -571,25 +526,102 @@ lookup_rel_int(dfs_t *dfs, dfs_obj_t *parent, const char *name, int flags, dfs_o
 out:
 	obj->flags = flags;
 	*_obj      = obj;
-
 	return rc;
 err_obj:
 	D_FREE(obj);
 	return rc;
 }
 
+int
+dfs_lookup(dfs_t *dfs, const char *path, int flags, dfs_obj_t **_obj, mode_t *mode,
+	   struct stat *stbuf)
+{
+	char *rem;
+	int   rc;
+
+	if (dfs == NULL || !dfs->mounted)
+		return EINVAL;
+	if (flags & O_APPEND)
+		return ENOTSUP;
+	if (_obj == NULL)
+		return EINVAL;
+	if (path == NULL || strnlen(path, DFS_MAX_PATH) > DFS_MAX_PATH - 1)
+		return EINVAL;
+	if (path[0] != '/')
+		return EINVAL;
+
+	/** if we added a prefix, check and skip over it */
+	if (dfs->prefix) {
+		if (strncmp(dfs->prefix, path, dfs->prefix_len) != 0)
+			return EINVAL;
+
+		path += dfs->prefix_len;
+	}
+
+	D_STRNDUP(rem, path, DFS_MAX_PATH - 1);
```

Comment:
Not sure to understand why rem needs to be duplicated... is it for being sure to not have a too long path ?

Reply:
nothing changed here from original implementation really, just moved.
the dup is needed because you cannot do strtok_r on a const string